academy/3-ibc/3-channels.md

```diff
@@ -67,17 +67,17 @@ Crossing hellos have been removed from ibc-go v4 onward, as referenced in [this
 
 </HighlightBox>
 
-Similarly, there are also [callbacks for the closing of channels](https://github.com/cosmos/ibc-go/blob/v5.1.0/modules/core/05-port/types/module.go#L71-L81). However, it is up to the application developers to decide if it makes sense for users to be able to trigger the closing of a channel. For token transfers for example, this is not desirable due to the risk of locking liquidity on a closed channel, hence that is [why this feature has been disallowed in the canonical `transfer` module](https://github.com/cosmos/ibc-go/blob/v5.1.0/modules/apps/transfer/ibc_module.go#L146-L153).
+Similarly, there are also [callbacks for the closing of channels](https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics#closing-handshake). However, it is up to the application developers to decide if it makes sense for users to be able to trigger the closing of a channel. For token transfers for example, this is not desirable due to the risk of locking liquidity on a closed channel, hence that is [why this feature has been disallowed in the canonical `transfer` module](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/apps/transfer/ibc_module.go#L151).
 
 ## Example code: `ChannelOpenInit`
 
-You can find the implementation of `ChannelOpenInit` in the the [`msg_server.go`](https://github.com/cosmos/ibc-go/blob/v5.1.0/modules/core/keeper/msg_server.go)
+You can find the implementation of `ChannelOpenInit` in core IBC's [`msg_server.go`](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/keeper/msg_server.go#L166)
 
 The important part to note in this code snippet is that an application module has capabilities for the requested port. Therefore, an application module can only use a channel and port if the application owns the capability for that port and the module which attempting to open a channel is the module we have granted capabilities to in `app.go`:
 
 <!-- TODO: add more content on capabilites based on Colin's deep dive https://github.com/cosmos/sdk-tutorials/issues/1277-->
 
-```go
+```go [https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/keeper/msg_server.go#L166]
```

Comment:
Why is a hyperlink needed here?

Reply:
With this, the code snippet has a link beneath, which leads directly to the source code whence it is taken.